scripts/quests/windurst/Class_Reunion.lua

```diff
@@ -0,0 +1,207 @@
+
+        [xi.zone.WINDURST_WALLS] =
+        {
+            ['_6n2'] = quest:progressEvent(413),
+
+                onEventFinish =
+                {
+                    [413] = function(player, csid, option, npc)
+                        quest:begin(player)
+                        npcUtil.giveKeyItem(player, xi.ki.CARBUNCLES_TEAR)
+                        quest:setVar(player, 'Prog', 1)
+                    end
+                },
+        },
+    },
+
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        --Section: Quest Accepted
+
+        [xi.zone.WINDURST_WALLS] =
+        {
+
+            ['Koru-Moru'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 1 then
+                        return quest:progressEvent(412)
+                    elseif questProgress == 2 then
+                        return quest:progressEvent(414)
+                    end
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if npcUtil.tradeHasExactly(trade, xi.items.ASTRAGALOS) then
+                        return quest:progressEvent(407)
+                    end
+                end,
+            },
+
+            ['Shantotto'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 3 then
+                        return quest:event(409)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [412] = function(player, csid, option, npc)
+                    player:delKeyItem(player, xi.ki.CARBUNCLES_TEAR)
+                    player:messageSpecial(ID.text.LOST_KEYITEM, xi.ki.CARBUNCLES_TEAR)
+                    quest:setVar(player, 'Prog', 2)
+                end,
+
+                [407] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                end,
+
+                [410] = function(player, csid, option, npc)
+                    quest:complete(player)
+                end,
+            }
+        },
+
+        [xi.zone.WINDURST_WATERS] =
+        {
+            ['Fuepepe'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 3 then
+                        return quest:event(817)
+                    end
+                end,
+            },
+
+            ['Furakku-Norakku'] =
+            {
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 3 then
+                        return quest:event(816)
+                    end
```

Comment:
```suggestion
                    if quest:getVar(player, 'Prog') == 3 then
                        return quest:event(816)
                    end
```

Reply:
What exactly issue the issue with this? The original does the same thing, and it's kinda just muscle memory to add that as some NPC are required multiple times for multiple different stages of tge quest?

Reply:
This isn't such a big deal. But doing it when it's only used in one spot is like saying:
 x = 1, y = 2 
 if x + y == 3 then ...
 
rather than saying 
if 1 + 2 == 3 then... 

Just boils down to code style really. It's always a good idea to cut down on lines of code when possible to boost readability in the future.

However, I'm sure as you're aware, if we need to call this memory location multiple times in a code block it's more efficient to store it once and prevent multiple memory look ups.